get-changed-files-from-env.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+const isFileIgnored = require('./is-file-ignored');
+
+/**
+ * Github actions jobs can run the following steps to get a fully accurate
+ * changed files list. Otherwise, we fallback to a simple diff between the
+ * current and base branch, which might give false positives if the base
+ * is ahead of the current branch.
+ *
+ *   - name: Get All Changed Files
+ *     uses: jaredly/get-changed-files@absolute
+ *     id: changed
+ *     with:
+ *       format: 'json'
+ *       absolute: true
+ *
+ *   - uses: allenevans/set-env@v2.0.0
+ *     with:
+ *       ALL_CHANGED_FILES: '${{ steps.changed.outputs.added_modified }}'
+ */
+const getChangedFilesFromEnv = (cwd /*:string*/) /*: ?Array<string>*/ => {
+    if (process.env.ALL_CHANGED_FILES) {
+        const files /*: Array<string> */ = JSON.parse(process.env.ALL_CHANGED_FILES); // flow-uncovered-line
+        return files.filter(path => !isFileIgnored(cwd, path));
+    }
+};
```

Comment:
What do we want the fail condition to be here? It seems like we may want to throw an error? Or log an error and return an empty array?

Reply:
If we expect people to be using this method directly then we should have error handling, but if we're only ever using it in the case of `getChangedFiles` then maybe we just have error logging?